File: tests/src/main/java/com/abstratt/simon/examples/UI2.java
package com.abstratt.simon.examples;

import com.abstratt.simon.metamodel.dsl.Meta;

@Meta.Package(builtIns = {})
public interface UI2 {
    class Form extends UI.Container {
        private UI.Link helpLink;

        @Meta.Contained
        public UI.Link helpLink() {
            return helpLink;
        }
    }
}

File: example-languages-kotlin/src/main/kotlin/com/abstratt/simon/examples/kotlin/UI2.kt
package com.abstratt.simon.examples.kotlin

import com.abstratt.simon.metamodel.dsl.Meta

@Meta.Package(builtIns = [])
interface UI2 {
    class Form : UI.Container() {
        private val helpLink: UI.Link? = null
        @Meta.Contained
        fun helpLink(): UI.Link? {
            return helpLink
        }
    }
}